### ISSUE TITLE
TouchableOpacity doesn't work on Android

### ISSUE THREAD

**Issue by clohr-vc** [kmagiera/react-native-gesture-handler#2440]
### Description

`TouchableOpacity` works fine on iOS, but doesn't seem to react to any inputs under Android.

### Steps to reproduce

1. Check out the linked Snack
2. Run on iOS emulator and verify that you can tap the red button saying "Tap me"
3. Now switch to Android emulator - you will find that the button is now unresponsive
4. Optional: Run on your native Android device - same result

### Snack or a link to a repository

https://snack.expo.dev/ol66JlC_h

### Gesture Handler version

2.8.0

### React Native version

0.70

### Platforms

Android, iOS

### JavaScript runtime

None

### Workflow

Expo managed workflow

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

Android emulator

### Acknowledgements

Yes

**Comment by clohr-vc** [comment]
I found a few other issues regarding similar behaviour, both on here and on StackOverflow (most famously https://github.com/software-mansion/react-native-gesture-handler/issues/578 ), but they all seem to be stemming from more complex setups (modals, the ability to simply switch imports towards `react-native` which we don't really have at the moment, absolute positioning, etc...).

The snack I provided aims to be as simplistic as possible, with a working button based on `TouchableOpacity` in both web and iOS, but failing in the Android environment.

The docs don't mention any Android specific setup code anymore, apart from the root wrapper needed for modals, so I'm not sure what else I'm missing. 

Any help is appreciated.
